### DISCUSSION TITLE
Reading a large .vtu mesh error (returning an empty mesh)

### DISCUSSION THREAD

**Discussion by Vijay47Kumar** [pyvista/pyvista#3586] [Q&A]
I am trying to read a 7GB vtu file and I get an error
"Cannot read cell offsets from Cells in piece 0 because the "offsets" array is not long enough."

And the mesh returned is empty. When doing this with a samller file from the same simulation with a courser mesh it reads in fine. Has anyone had this problem? 

![image](https://user-images.githubusercontent.com/45141958/201339299-1ef384ee-0d76-42f7-89e1-2d29a0c42e97.png)

![image](https://user-images.githubusercontent.com/45141958/201339659-2b69a07b-b166-4f30-84f8-45088fbe7104.png)

**Comment by banesullivan** [comment]
It's going to be very tough to uncover what's going wrong without the file in question... which is going to be tough to transfer. Can you download ParaView and see if it opens with ParaView? https://www.paraview.org/download/

**Reply by MatthewFlamm** [reply]
There are lots of reports like this on the VTK and ParaView forums (and some CFD forums).  I'll leave a few here.  Hopefully it can help.  https://discourse.paraview.org/t/problem-reading-large-vtu-file/6887.  https://gitlab.kitware.com/paraview/paraview/-/issues/19628

**Reply by MatthewFlamm** [reply]
One thing noted by one report is that removing data led to successful reading.  You could try turning off reading the data in the reader.  I'm not sure whether this will work the same as removing it from the file however:

```py
reader = pv.get_reader("file.vtu")
reader.disable_all_cell_arrays()
reader.disable_all_point_arrays()
reader.enable_cell_array("velocity")  # an example of just reading in one piece of data
mesh=reader.read()
```
